### PR TITLE
Migrate to .NET 8 Collection Expressions

### PR DESCRIPTION
**Pull request by IsaacMarovitz**
This is a re-tool of part of #5928. 

- Enables `IDE0300`, and `IDE0301`
- `IDE0028`, `IDE0302`, `IDE0305` are left disabled, as the resulting code is functionally different in a way that could harm performance and/or readability

### CODE REVIEW

File: src/ARMeilleure/Instructions/InstEmitSimdHelper.cs [hunk 1]
```diff
@@ -44,118 +44,118 @@ public static ulong X86GetGf2p8LogicalShiftLeft(int shift)
...
-        public static readonly Intrinsic[] X86PaddInstruction = new Intrinsic[]
-        {
+        public static readonly Intrinsic[] X86PaddInstruction =
+        [
             Intrinsic.X86Paddb,
             Intrinsic.X86Paddw,
             Intrinsic.X86Paddd,
-            Intrinsic.X86Paddq,
-        };
+            Intrinsic.X86Paddq
```
Comment (AcK77) [inline src/ARMeilleure/Instructions/InstEmitSimdHelper.cs:52]: Ident is off and `,` is missing

(Same for other changes in this file)

File: src/ARMeilleure/Translation/PTC/PtcProfiler.cs [hunk 2]
```diff
@@ -25,9 +25,10 @@ class PtcProfiler
 
         private const uint InternalVersion = 5518; //! Not to be incremented manually for each change to the ARMeilleure project.
 
-        private static readonly uint[] _migrateInternalVersions = {
-            1866,
-        };
+        private static readonly uint[] _migrateInternalVersions =
+        [
+            1866
```
Comment (AcK77) [inline src/ARMeilleure/Translation/PTC/PtcProfiler.cs:30]: `,` is missing

File: src/ARMeilleure/Translation/Translator.cs [hunk 3]
```diff
@@ -23,24 +23,22 @@ namespace ARMeilleure.Translation
...
-                new(15,  8),
-                new( 7,  8),
-                new( 2,  5),
-            };
+        [
+            new(31, 17),
+            new(23, 8),
+            new(15, 8),
+            new(7, 8),
+            new(2, 5)
```
Comment (AcK77) [inline src/ARMeilleure/Translation/Translator.cs:31]: `,` is missing

File: src/ARMeilleure/Translation/Translator.cs [hunk 4]
```diff
@@ -23,24 +23,22 @@ namespace ARMeilleure.Translation
...
-                new(15,  8),
-                new( 7,  8),
-                new( 1,  6),
-            };
+        [
+            new(31, 17),
+            new(23, 8),
+            new(15, 8),
+            new(7, 8),
+            new(1, 6)
```
Comment (AcK77) [inline src/ARMeilleure/Translation/Translator.cs:40]: `,` is missing

File: src/Ryujinx.Audio/Backends/CompatLayer/Downmixing.cs [hunk 5]
```diff
@@ -31,19 +31,19 @@ private struct ChannelStereoFormatPCM16
...
-        private static readonly long[] _defaultSurroundToStereoCoefficients = new long[4]
-        {
+        private static readonly long[] _defaultSurroundToStereoCoefficients =
+        [
             RawQ15One,
             Minus3dBInQ15,
             Minus12dBInQ15,
-            Minus3dBInQ15,
-        };
+            Minus3dBInQ15
```
Comment (AcK77) [inline src/Ryujinx.Audio/Backends/CompatLayer/Downmixing.cs:39]: `,` is missing

File: src/Ryujinx.Audio/Backends/CompatLayer/Downmixing.cs [hunk 6]
```diff
@@ -31,19 +31,19 @@ private struct ChannelStereoFormatPCM16
...
+        ];
 
-        private static readonly long[] _defaultStereoToMonoCoefficients = new long[2]
-        {
-            Minus6dBInQ15,
+        private static readonly long[] _defaultStereoToMonoCoefficients =
+        [
             Minus6dBInQ15,
-        };
+            Minus6dBInQ15
```
Comment (AcK77) [inline src/Ryujinx.Audio/Backends/CompatLayer/Downmixing.cs:45]: `,` is missing

File: src/Ryujinx.Audio/Constants.cs [hunk 7]
```diff
@@ -164,12 +164,12 @@ public static class Constants
...
-        public static readonly float[] DefaultSurroundToStereoCoefficients = new float[4]
-        {
+        public static readonly float[] DefaultSurroundToStereoCoefficients =
+        [
             1.0f,
             0.707f,
             0.251f,
-            0.707f,
-        };
+            0.707f
```
Comment (AcK77) [inline src/Ryujinx.Audio/Constants.cs:172]: `,` is missing

File: src/Ryujinx.Audio/Renderer/Device/VirtualDevice.cs [hunk 8]
```diff
@@ -10,14 +10,14 @@ public class VirtualDevice
...
-        {
+        public static readonly VirtualDevice[] Devices =
+        [
             new("AudioStereoJackOutput", 2, true),
             new("AudioBuiltInSpeakerOutput", 2, false),
             new("AudioTvOutput", 6, false),
             new("AudioUsbDeviceOutput", 2, true),
-            new("AudioExternalOutput", 6, true),
-        };
+            new("AudioExternalOutput", 6, true)
```
Comment (AcK77) [inline src/Ryujinx.Audio/Renderer/Device/VirtualDevice.cs:19]: `,` is missing

**Review by AcK77 — CHANGES_REQUESTED** [review]

File: src/ARMeilleure/Instructions/InstEmitSimdHelper.cs (hunk 1 above)
Comment (IsaacMarovitz) [reply src/ARMeilleure/Instructions/InstEmitSimdHelper.cs:52]: I didn't bother keeping the last `,` if it didn't affect the indentation of comments. If you really want me to keep them I can add them back in.

Comment (AcK77) [reply src/ARMeilleure/Instructions/InstEmitSimdHelper.cs:52]: That’s something we usually do in our codebase since we added auto-formatting

**Pull request closed without merging**